x-pack/plugins/observability_solution/apm/public/components/app/diagnostics/index.tsx

```diff
@@ -10,26 +10,44 @@ import React from 'react';
 import * as t from 'io-ts';
 import { EuiButton, EuiCallOut, EuiIcon, EuiLoadingLogo, EuiEmptyPrompt } from '@elastic/eui';
 import { i18n } from '@kbn/i18n';
+import { dynamic } from '@kbn/shared-ux-utility';
 import { useApmParams } from '../../../hooks/use_apm_params';
 import { useApmRouter } from '../../../hooks/use_apm_router';
 import { useApmRoutePath } from '../../../hooks/use_apm_route_path';
-import { DiagnosticsSummary, getIsCrossCluster } from './summary_tab';
-import { ApmMainTemplate } from '../../routing/templates/apm_main_template';
-import { DiagnosticsIndexTemplates } from './index_templates_tab';
-import { DiagnosticsIndices } from './indices_tab';
-import { DiagnosticsDataStreams } from './data_stream_tab';
-import {
-  DiagnosticsIndexPatternSettings,
-  getIsIndexPatternTabOk,
-} from './index_pattern_settings_tab';
-import { DiagnosticsImportExport } from './import_export_tab';
+import { getIsCrossCluster } from './summary_tab';
+import { getIsIndexPatternTabOk } from './index_pattern_settings_tab';
 import { DiagnosticsContextProvider } from './context/diagnostics_context';
 import { useDiagnosticsContext } from './context/use_diagnostics';
 import { getIsIndexTemplateOk } from './summary_tab/index_templates_status';
 import { getIsDataStreamTabOk } from './summary_tab/data_streams_status';
 import { getIsIndicesTabOk } from './summary_tab/indicies_status';
-import { DiagnosticsApmDocuments } from './apm_documents_tab';
 import { isPending } from '../../../hooks/use_fetcher';
+import { ApmMainTemplate } from '../../routing/templates/apm_main_template';
+
+const DiagnosticsSummary = dynamic(() =>
+  import('./summary_tab').then((mod) => ({ default: mod.DiagnosticsSummary }))
+);
+
+const DiagnosticsIndexTemplates = dynamic(() =>
+  import('./index_templates_tab').then((mod) => ({ default: mod.DiagnosticsIndexTemplates }))
+);
+const DiagnosticsIndices = dynamic(() =>
+  import('./indices_tab').then((mod) => ({ default: mod.DiagnosticsIndices }))
+);
+const DiagnosticsDataStreams = dynamic(() =>
+  import('./data_stream_tab').then((mod) => ({ default: mod.DiagnosticsDataStreams }))
+);
+const DiagnosticsIndexPatternSettings = dynamic(() =>
+  import('./index_pattern_settings_tab').then((mod) => ({
+    default: mod.DiagnosticsIndexPatternSettings,
+  }))
+);
+const DiagnosticsImportExport = dynamic(() =>
+  import('./import_export_tab').then((mod) => ({ default: mod.DiagnosticsImportExport }))
+);
+const DiagnosticsApmDocuments = dynamic(() =>
+  import('./apm_documents_tab').then((mod) => ({ default: mod.DiagnosticsApmDocuments }))
+);
```

Comment:
IMHO we don't need to go overboard here. Can't we load the diagnostics page in a single async chunk?

Reply:
Not prob, I can revert this to a single chunk removing the imports. To keep in mind, this will load an extra ~10kb of minified code before it gets really used in that async chunk, which is not a big deal but still worth noticing.

Reply:
A user landing on the diagnostics page loads the JS for all the available tabs at once, even if the user only visits the Summary tab. With the lazy loading I used, every tab chunk (~2.5kb) is loaded only when visiting the tab. Should we still keep it split in this case?

Reply:
nah let's just do one chunk, 2.5kb is not worth the network overhead

Reply:
https://github.com/user-attachments/assets/c6e4f9b8-be73-404e-bab6-8cda63c33ca2

Reply:
btw, same for the others, I think we can find a better balance between loading too much upfront and loading too many small chunks

Reply:
I reduced code splitting to where it only has the biggest impact now. The entry point for other pages is not impacted (load only apm.plugin.js)